docs/API.md

```diff
@@ -0,0 +1,507 @@
+```
+
+### Discussion:
+Creates a child logger, setting all key-value pairs in `bindings` as properties
+in the log lines. All serializers will be applied to the given pair.
+
+Child loggers use the same output stream as the parent and inherit
+the current log level of the parent at the time they are spawned.
+
+From v2.x.x the log level of a child is mutable (whereas in
+v1.x.x it was immutable), and can be set independently of the parent.
+If a `level` property is present in the object passed to `child` it will
+override the child logger level.
+
+For example:
+
+```
+var logger = pino()
+logger.level = 'error'
+logger.info('nope') //does not log
+var child = logger.child({foo: 'bar'})
+child.info('nope again') //does not log
+child.level = 'info'
+child.info('hooray') //will log
+logger.info('nope nope nope') //will not log, level is still set to error
+logger.child({ foo: 'bar', level: 'debug' }).debug('debug!')
+```
+
+Child loggers inherit the serializers from the parent logger but it is
+possible to override them.
+
+For example:
+
+```
+var pino = require('./pino')
+
+var customSerializers = {
+  test: function () {
+    return 'this is my serializer'
+  }
+}
+var child = pino().child({serializers: customSerializers})
+
+child.info({test: 'should not show up'})
+```
+
+Will produce the following output:
+
+```
+{"pid":7971,"hostname":"mycomputer.local","level":30,"time":1469488147985,"test":"this is my serializer","v":1}
+```
+
+Also from version 2.x.x we can spawn child loggers from child loggers, for instance:
+
+```
+var logger = pino()
+var child = logger.child({father: true})
+var childChild = child.child({baby: true})
+```
+
+Child logger creation is fast:
+
+```
+benchBunyanCreation*10000: 1291.332ms
+benchBoleCreation*10000: 1630.542ms
+benchPinoCreation*10000: 352.330ms
+benchPinoExtremeCreation*10000: 102.282ms
+```
+
+Logging through a child logger has little performance penalty:
+
+```
+benchBunyanChild*10000: 1343.933ms
+benchBoleChild*10000: 1605.969ms
+benchPinoChild*10000: 334.573ms
+benchPinoExtremeChild*10000: 152.792ms
+```
+
+Spawning children from children has negligible overhead:
+
+```
+benchBunyanChildChild*10000: 1397.202ms
+benchPinoChildChild*10000: 338.930ms
+benchPinoExtremeChildChild*10000: 150.143ms
+```
+
+<a id="level"></a>
+## .level
+
+### Example:
+```
+logger.level = 'info'
+```
+
+### Discussion:
+
+Set this property to the desired logging level. In order of priority, available
+levels are:
+
+  1. <a href="#fatal">`'fatal'`</a>
+  2. <a href="#error">`'error'`</a>
+  3. <a href="#warn">`'warn'`</a>
+  4. <a href="#info">`'info'`</a>
+  5. <a href="#debug">`'debug'`</a>
+  6. <a href="#trace">`'trace'`</a>
+
+The logging level is a *minimum* level. For instance if `logger.level` is
+`'info'` then all `'fatal'`, `'error'`, `'warn'`, and `'info'` logs will be enabled.
+
+You can pass `'silent'` to disable logging.
+
+<a id="fatal"></a>
```

Comment:
In my opinion, it was a stupid deprecation. Absolutely no one had a problem with the way things worked, and they still expect it to work that way today.